sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/models/KeyVaultCertificateIdentifier.java

```diff
@@ -16,11 +16,47 @@
 public final class KeyVaultCertificateIdentifier {
     private final String certificateId, vaultUrl, name, version;
 
-    private KeyVaultCertificateIdentifier(String certificateId, String vaultUrl, String name, String version) {
-        this.certificateId = certificateId;
-        this.vaultUrl = vaultUrl;
-        this.name = name;
-        this.version = version;
+    /**
+     * Create a new {@link KeyVaultCertificateIdentifier} from a given certificate identifier.
+     *
+     * <p>Valid examples are:
+     *
+     * <ul>
+     *     <li>https://{key-vault-name}.vault.azure.net/certificates/{certificate-name}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/certificates/{certificate-name}/pending</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/certificates/{certificate-name}/{unique-version-id}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/deletedcertificates/{deleted-certificate-name}</li>
+     * </ul>
+     *
+     * @param certificateId The certificate identifier to extract information from.
+     *
+     * @throws IllegalArgumentException if the given identifier is {@code null} or an invalid Key Vault Certificate
+     * identifier.
+     */
+    public KeyVaultCertificateIdentifier(String certificateId) {
+        if (certificateId == null) {
+            throw new IllegalArgumentException("certificateId cannot be null");
+        }
+
+        try {
+            final URL url = new URL(certificateId);
+            // We expect an identifier with either 2 or 3 path segments: collection + name [+ version]
+            final String[] pathSegments = url.getPath().split("/");
+
+            if ((pathSegments.length != 3 && pathSegments.length != 4) // More or less segments in the URI than expected.
+                || !"https".equals(url.getProtocol()) // Invalid protocol.
+                || (!"certificates".equals(pathSegments[1]) && !"deletedcertificates".equals(pathSegments[1])) // Invalid collection.
```

Comment:
Do you think we should rename the class to `KeyVaultIdentifier` then? Otherwise, I think other languages should verify that the collection is correct, else it could be confusing that a class made specifically for Certificates accepts other collections.

Reply:
We wanted separate names to make sure there wasn't any collisions. It's not uncommon to have multiple packages imported simultaneously.

Reply:
.NET, JS, and Python don't assert collections (just double checked). Probably best not to here either.